File: investments/src/main/java/com/abm/investments/resources/InvestmentResource.java
package com.abm.investments.resources;

import com.abm.investments.dtos.InvestmentDto;
import com.abm.investments.model.Amount;
import com.abm.investments.model.Investment;
import com.abm.investments.repositories.InvestmentRepository;
import org.springframework.beans.factory.annotation.Autowired;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("investments")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class InvestmentResource {

    @Autowired
    private InvestmentRepository investmentRepository;

    @GET
    public Response getAll() {
        return Response.ok(investmentRepository.findAll()).build();
    }

    @POST
    public Response create(InvestmentDto investmentDto) {
        Investment investment = new Investment(investmentDto.getName(), new Amount(investmentDto.getAmount()));
        return Response.ok(investmentRepository.save(investment)).build();
    }

    @PUT
    public void update(Investment investment) {
        investmentRepository.save(investment);
    }

}

File: property-manager/src/main/scala/com/propertymanager/resources/InvestmentResource.scala
package com.propertymanager.resources

import java.io.IOException
import javax.ws.rs.core.{MediaType, Response}
import javax.ws.rs.{Consumes, POST, Path, Produces}

import com.propertymanager.domain.Investment
import com.propertymanager.services.InvestmentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
@Path("investments")
@Consumes(Array(MediaType.APPLICATION_JSON))
@Produces(Array(MediaType.APPLICATION_JSON))
class InvestmentResource(@Autowired val investmentService: InvestmentService) {

  @POST
  def create(investment: Investment): Response = {
    try {
      Response.ok(investmentService.add(investment)).build()
    } catch {
      case ex: IOException => {
        Response.status(Response.Status.BAD_REQUEST).build()
      }
      case ex: Exception => {
        Response.serverError().build()
      }
    }


  }

}
